### PR TITLE
Addition of bucketclass edit

### PR DESCRIPTION
**Pull request by jeniawhite**
### Explain the changes
1. Added option of bucketclass editing.
This will edit all of the tiering_policies and tiers for the relevant buckets

### Issues: Fixed #xxx / Gap #xxx
1. 

### Testing Instructions:
1.

### CODE REVIEW

File: src/api/tiering_policy_api.js [hunk 1]
```diff
@@ -139,6 +172,18 @@ module.exports = {
 
     definitions: {
 
+        data_placement_enum: {
```
Comment (nimrod-becker) [inline src/api/tiering_policy_api.js]: Already defined in tier_api, just use it from there

File: src/api/tiering_policy_api.js [hunk 2]
```diff
@@ -139,6 +172,18 @@ module.exports = {
...
+            enum: ['MIRROR', 'SPREAD'],
+            type: 'string',
+        },
+
+        pool_info: {
```
Comment (nimrod-becker) [inline src/api/tiering_policy_api.js]: Also defined in tier_api

File: src/server/system_services/tier_server.js [hunk 3]
```diff
@@ -295,6 +296,62 @@ function update_policy(req) {
...
+    } catch (error) {
+        dbg.error('_check_validity_of_configuration failed with', error);
+        throw new RpcError('BAD_REQUEST', 'invalid attached_pools in tiers');
+    }
+}
+
+async function update_bucket_class(req) {
+    const { name, policy, tiers } = req.rpc_params;
+    if (policy.tiers.length !== tiers.length) throw new RpcError('BAD_REQUEST', 'policy tiers and tiers length mismatch');
+    const bucket_class_buckets = _.filter(system_store.data.buckets, bucket =>
```
Comment (nimrod-becker) [inline src/server/system_services/tier_server.js]:
```suggestion
    //Filter only buckets created via OBC
    const bucket_class_buckets = _.filter(system_store.data.buckets, bucket =>
```

File: src/server/system_services/tier_server.js [hunk 4]
```diff
@@ -295,6 +296,62 @@ function update_policy(req) {
...
+        bucket.bucket_claim && (_.isEqual(bucket.bucket_claim.bucket_class, name)));
+    if (_.isEmpty(bucket_class_buckets)) return;
+    _check_validity_of_configuration({ req, bucket_class_buckets });
+    for (const bucket of bucket_class_buckets) {
+        const old_policy = bucket.tiering;
+        const old_tiers = old_policy.tiers
+            .map(obj => obj.tier.name);
+        if (tiers.length < old_tiers.length) throw new RpcError('BAD_REQUEST', 'cannot remove tiers');
```
Comment (nimrod-becker) [inline src/server/system_services/tier_server.js]:
```suggestion
```

Already performed in validation

**Review by nimrod-becker — APPROVED** [review]

**Review by dannyzaken — APPROVED** [review]

**Pull request merged**